hips/tiles/fetch.py

```diff
@@ -0,0 +1,120 @@
+        self.tile_indices = tile_indices
+        self.hips_order = hips_order
+        self.hips_survey = hips_survey
+        self.tile_format = tile_format
+        self.progress_bar = progress_bar
+        self.thread_count = thread_count
+
+    def fetch_tile_threaded(self, url: str, timeout: int = 10) -> Generator:
+        """Fetch a HiPS tile asynchronously."""
+        with urllib.request.urlopen(url, timeout=timeout) as conn:
+            return conn.read()
+
+    @property
+    def tiles(self) -> np.ndarray:
+        """Generator function to fetch HiPS tiles from a remote URL."""
+        tile_urls, tile_metas = [], []
+        for healpix_pixel_index in self.tile_indices:
+            tile_meta = HipsTileMeta(
+                order=self.hips_order,
+                ipix=healpix_pixel_index,
+                frame=self.hips_survey.astropy_frame,
+                file_format=self.tile_format,
+            )
+            tile_urls.append(self.hips_survey.tile_url(tile_meta))
+            tile_metas.append(tile_meta)
+
+        raw_responses = []
+        with concurrent.futures.ThreadPoolExecutor(max_workers=self.thread_count) as executor:
+            future_to_url = {executor.submit(self.fetch_tile_threaded, url): url for url in tile_urls}
+            if self.progress_bar:
+                from tqdm import tqdm
+                requests = tqdm(concurrent.futures.as_completed(future_to_url), total=len(future_to_url), desc='Fetching tiles')
+            else:
+                requests = concurrent.futures.as_completed(future_to_url)
+
+            for future in requests:
+                url = future_to_url[future]
+                raw_responses.append(future.result())
+
+        tiles = []
+        for idx, raw_data in enumerate(raw_responses):
+            tiles.append(HipsTile(tile_metas[idx], raw_data))
+        return tiles
+
+    async def fetch_tile_threaded_aiohttp(self, url: str, session) -> Generator:
+        """Fetch a HiPS tile asynchronously using aiohttp."""
+        async with session.get(url) as response:
+            return await response.read()
+
+    @property
+    async def fetch_tiles_aiohttp(self) -> np.ndarray:
+        """Generator function to fetch HiPS tiles from a remote URL using aiohttp."""
+        import aiohttp, asyncio
+        tile_urls, tile_metas = [], []
+        for healpix_pixel_index in self.tile_indices:
+            tile_meta = HipsTileMeta(
+                order=self.hips_order,
+                ipix=healpix_pixel_index,
+                frame=self.hips_survey.astropy_frame,
+                file_format=self.tile_format,
+            )
+            tile_urls.append(self.hips_survey.tile_url(tile_meta))
+            tile_metas.append(tile_meta)
+
+        tasks = []
+        async with aiohttp.ClientSession() as session:
+            for idx, url in enumerate(tile_urls):
+                task = asyncio.ensure_future(self.fetch_tile_threaded_aiohttp(url.format(idx), session))
```

Comment:
I see your `fetch_tiles_aiohttp` calls a method called `fetch_tile_threaded_aiohttp`.
async doesn't use threads, it runs in a single thread.
Maybe just rename the method to be accurate?